`scikit-learn` release breaks quality tests

`scikit-learn 1.1.0` included the following update: https://github.com/scikit-learn/scikit-learn/pull/21026.

This new feature causes an error in our quality tests, with the following stack trace:

```
def test_quality(subtests):
        """Run all the quality test cases.
    
        This test has multiple steps.
            1. It creates a list of test cases. Each test case has a dataset
            and a set of sdtypes to test for the dataset.
            2. A dictionary is created mapping sdtypes to a DataFrame
            containing the regression scores obtained from running the
            transformers of that sdtype against the datasets in the test cases.
            Each row in the DataFrame has the transformer name, dataset name,
            column name and score. The scores are computed as follows:
                - For every transformer of the sdtype, transform all the
                columns of that sdtype.
                - For every numerical column in the dataset, the transformed
                columns are used as features to train a regression model.
                - The score is the coefficient of determination obtained from
                that model trying to predict the target column.
            3. Once the scores are gathered, a results table is created. Each row has
            a transformer name, dataset name, average score for the dataset and a score
            comparing the transformer's average score for the dataset to the average
            of the average score for the dataset across all transformers of the same
            sdtype.
            4. For every unique transformer in the results, a test is run to check
            that the transformer's score for each table is either higher than the
            threshold, or the comparitive score is higher than the threshold.
        """
        transformers_by_type = get_transformers_by_type()
        sdtypes_to_test = {
            sdtype
            for sdtype in transformers_by_type.keys()
            if sdtype not in SDTYPES_TO_SKIP
        }
        test_cases = get_test_cases(sdtypes_to_test)
>       all_regression_scores = get_regression_scores(test_cases, transformers_by_type)

tests/quality/test_quality.py:[23](https://github.com/sdv-dev/RDT/runs/6414526841?check_suite_focus=true#step:5:23)2: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
tests/quality/test_quality.py:146: in get_regression_scores
    regression_scores = get_transformer_regression_scores(
tests/quality/test_quality.py:98: in get_transformer_regression_scores
    score = get_regression_score(transformed_features, target)
tests/quality/test_quality.py:40: in get_regression_score
    scores = cross_val_score(model, features, target)
/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/sklearn/model_selection/_validation.py:515: in cross_val_score
    cv_results = cross_validate(
/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/sklearn/model_selection/_validation.py:285: in cross_validate
    _warn_or_raise_about_fit_failures(results, error_score)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

results = [{'fit_error': 'Traceback (most recent call last):\n  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-...eError: Input y contains NaN.\n', 'fit_time': 0.000146150588989[25](https://github.com/sdv-dev/RDT/runs/6414526841?check_suite_focus=true#step:5:25)78, 'score_time': 0.0, 'test_scores': {'score': nan}}]
error_score = nan

    def _warn_or_raise_about_fit_failures(results, error_score):
        fit_errors = [
            result["fit_error"] for result in results if result["fit_error"] is not None
        ]
        if fit_errors:
            num_failed_fits = len(fit_errors)
            num_fits = len(results)
            fit_errors_counter = Counter(fit_errors)
            delimiter = "-" * 80 + "\n"
            fit_errors_summary = "\n".join(
                f"{delimiter}{n} fits failed with the following error:\n{error}"
                for error, n in fit_errors_counter.items()
            )
    
            if num_failed_fits == num_fits:
                all_fits_failed_message = (
                    f"\nAll the {num_fits} fits failed.\n"
                    "It is is very likely that your model is misconfigured.\n"
                    "You can try to debug the error by setting error_score='raise'.\n\n"
                    f"Below are more details about the failures:\n{fit_errors_summary}"
                )
>               raise ValueError(all_fits_failed_message)
E               ValueError: 
E               All the 5 fits failed.
E               It is is very likely that your model is misconfigured.
E               You can try to debug the error by setting error_score='raise'.
E               
E               Below are more details about the failures:
E               --------------------------------------------------------------------------------
E               5 fits failed with the following error:
E               Traceback (most recent call last):
E                 File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/sklearn/model_selection/_validation.py", line 686, in _fit_and_score
E                   estimator.fit(X_train, y_train, **fit_params)
E                 File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/sklearn/linear_model/_base.py", line 684, in fit
E                   X, y = self._validate_data(
E                 File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/sklearn/base.py", line 596, in _validate_data
E                   X, y = check_X_y(X, y, **check_params)
E                 File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/sklearn/utils/validation.py", line 1086, in check_X_y
E                   y = _check_y(y, multi_output=multi_output, y_numeric=y_numeric, estimator=estimator)
E                 File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/sklearn/utils/validation.py", line 1096, in _check_y
E                   y = check_array(
E                 File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/sklearn/utils/validation.py", line 895, in check_array
E                   _assert_all_finite(
E                 File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/sklearn/utils/validation.py", line 1[42](https://github.com/sdv-dev/RDT/runs/6414526841?check_suite_focus=true#step:5:42), in _assert_all_finite
E                   raise ValueError(msg_err)
E               ValueError: Input y contains NaN.
```

--- a/tests/quality/test_quality.py
+++ b/tests/quality/test_quality.py
@@ -89,12 +89,15 @@ def get_transformer_regression_scores(data, sdtype, dataset_name, transformers, 
         numerical_transformer = FloatFormatter(model_missing_values=False)
         target = numerical_transformer.fit_transform(target, column)
         target = format_array(target)
+        nans = np.isnan(target)[:, 0]
+        target = target[~nans]
         for transformer in transformers:
             ht = HyperTransformer()
             ht.detect_initial_config(features)
             ht.update_transformers_by_sdtype(sdtype=sdtype, transformer=transformer())
             ht.fit(features)
             transformed_features = ht.transform(features).to_numpy()
+            transformed_features = transformed_features[~nans]
             score = get_regression_score(transformed_features, target)
             row = pd.Series({
                 'transformer_name': transformer.__name__,

